lib/ocpp/v201/charge_point.cpp

```diff
@@ -425,6 +425,26 @@ void ChargePoint::init_websocket() {
 
     this->websocket = std::make_unique<Websocket>(connection_options, this->pki_handler, this->logging);
     this->websocket->register_connected_callback([this](const int security_profile) {
+        // stop the offline threshold timer
+        offline_threshold_timer.stop();
+
+        if (!is_offline_threshold_expired) {
+            EVLOG_debug << "LIBOCPP: Offline < threshold";
+            // get the status of all the connectors
+            for (auto const& [evse_id, evse] : this->evses) {
+                int number_of_connectors = evse->get_number_of_connectors();
+                EVLOG_debug << "LIBOCPP: evseId: " << evse_id << " numConn: " << number_of_connectors;
+                for (int connector_id = 1; connector_id <= number_of_connectors; connector_id++) {
+                    if (!(conn_state_per_evse[{evse_id,connector_id}].connector_status ==
+                          evse->get_state(connector_id))) {
+                        // if the state varied when offline
+                        conn_state_per_evse[{evse_id,connector_id}].is_status_changed_when_offline = true;
```

Comment:
Why do we need to store this value for use later instead of using it directly?

Reply:
What about just checking the registration status in the connection callback? When we are accepted we stay accepted even after a lost connection. If we were not accepted yet we can just drop the changes since we are required to send the status notifications of all connectors anyway after a BootNotificationResponse with accepted (B01.FR.05).